avm/res/network/public-ip-address/main.bicep

```
@@ -0,0 +1,280 @@
+metadata name = 'Public IP Addresses'
+metadata description = 'This module deploys a Public IP Address.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. The name of the Public IP Address.')
+param name string
+
+@description('Optional. Resource ID of the Public IP Prefix object. This is only needed if you want your Public IPs created in a PIP Prefix.')
+param publicIpPrefixResourceId publicIpPrefixType?
+
+@description('Optional. The public IP address allocation method.')
+@allowed([
+  'Dynamic'
+  'Static'
+])
+param publicIPAllocationMethod string = 'Static'
+
+@description('Optional. A list of availability zones denoting the IP allocated for the resource needs to come from.')
+param zones array?
+
+@description('Optional. IP address version.')
+@allowed([
+  'IPv4'
+  'IPv6'
+])
+param publicIPAddressVersion string = 'IPv4'
+
+@description('Optional. The DNS settings of the public IP address.')
+param dnsSettings dnsSettingsType?
+
+@description('Optional. The lock settings of the service.')
+param lock lockType
+
+@description('Optional. Name of a public IP address SKU.')
+@allowed([
+  'Basic'
+  'Standard'
+])
+param skuName string = 'Standard'
+
+@description('Optional. Tier of a public IP address SKU.')
+@allowed([
+  'Global'
+  'Regional'
+])
+param skuTier string = 'Regional'
+
+@description('Optional. The DDoS protection plan configuration associated with the public IP address.')
+param ddosSettings ddosSettingsType?
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Optional. Array of role assignment objects that contain the \'roleDefinitionIdOrName\' and \'principalId\' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or its fully qualified ID in the following format: \'/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11\'.')
+param roleAssignments roleAssignmentType?
```

```suggestion
param roleAssignments roleAssignmentType
```
The type is already nullable